README.md

```diff
@@ -1559,28 +1256,357 @@ let main ~config_changed =
-See Eio's own tests for examples, e.g., [tests/switch.md](tests/switch.md).
+### Executor Pool
 
-## Provider Interfaces
+An executor pool distributes jobs among a pool of domains workers (threads).
+Domains are reused and can execute multiple jobs concurrently.
+Jobs are queued up if they cannot be started immediately due to all workers being busy.
+The `Eio.Executor_pool` module is the recommended way of leveraging OCaml 5's multicore capabilities.
 
-Eio applications use resources by calling functions (such as `Eio.Flow.write`).
+Usually you will only want one pool for an entire application, so the pool is typically created when the application starts:
+
+<!-- $MDX skip -->
+```ocaml
+let () =
+  Eio_main.run @@ fun env ->
+  Switch.run @@ fun sw ->
+  let pool =
+    Eio.Executor_pool.create
+      ~sw (Eio.Stdenv.domain_mgr env)
+      ~domain_count:4
+  in
+  main ~pool
+```
+
+The pool starts its domain workers immediately upon creation.
+
+The pool will not block our switch `sw` from completing;
+when the switch finishes, all domain workers and running jobs are cancelled.
+
+`~domain_count` is the number of domain workers (threads) to create.
+This number should not exceed `Domain.recommended_domain_count` or the number of cores on your system.
+Additionally, consider reducing this number by 1 if, for example, your original domain will be performing CPU intensive work at the same time as the pool.
+
+Now let's say we have a CPU intensive task:
+
+```ocaml
+let sum_to n =
+  traceln "Starting CPU-intensive task...";
+  let total = ref 0 in
+  for i = 1 to n do
+    total := !total + i
+  done;
+  traceln "Finished";
+  !total
+```
+
+We can have it run in parallel using an Executor Pool:
+
+```ocaml
+let main ~domain_mgr =
+  Switch.run @@ fun sw ->
+  let pool =
+    Eio.Executor_pool.create ~sw domain_mgr ~domain_count:4
+  in
+  let test n =
+    traceln "sum 1..%d = %d" n
+      (Eio.Executor_pool.submit_exn pool ~weight:1.0
+        (fun () -> sum_to n))
+  in
+  Fiber.both
+    (fun () -> test 100000)
+    (fun () -> test 50000)
+```
+
+<!-- $MDX non-deterministic=output -->
+```ocaml
+# Eio_main.run @@ fun env ->
+  main ~domain_mgr:(Eio.Stdenv.domain_mgr env);;
++Starting CPU-intensive task...
++Starting CPU-intensive task...
++Finished
++sum 1..50000 = 1250025000
++Finished
++sum 1..100000 = 5000050000
+- : unit = ()
+```
+Note: the exact `traceln` output of this example is non-deterministic, because the OS is free to schedule domains as it likes.
+
+`~weight` is the anticipated proportion of a CPU core used by the job.
+Here we use `~weight:1.0` because the job is entirely CPU-bound: it never waits for I/O or other syscalls.
+`~weight` must be `>= 0.0` and `<= 1.0`;
+Example: given an IO-bound job that averages 2% of one CPU core, pass `~weight:0.02`.
+
+Each domain worker starts new jobs until the total `~weight` of its running jobs reaches `1.0`.
+
+### Domain Manager
```

Comment:
Perhaps move the Domain Manager part on top and then introduce the Executor pool as an improvement?

Reply:
I'd prefer to keep the docs on the Executor Pool self-sufficient so users don't have to learn about a different module to understand it. I've made small tweaks to help with the confusion you've encountered.